src/workers/transactions/userReceivedAward.ts

```diff
@@ -0,0 +1,92 @@
+import { queryReadReplica } from '../../common/queryReadReplica';
+import { Feature, FeatureType, UserPost } from '../../entity';
+import { UserComment } from '../../entity/user/UserComment';
+import {
+  UserTransaction,
+  UserTransactionProcessor,
+} from '../../entity/user/UserTransaction';
+import { NotificationType } from '../../notifications/common';
+import { generateTypedNotificationWorker } from '../notifications/worker';
+
+export const userReceivedAward =
+  generateTypedNotificationWorker<'api.v1.user-transaction'>({
+    subscription: 'api.user-received-award',
+    handler: async (data, con, logger) =>
+      queryReadReplica(con, async ({ queryRunner }) => {
+        const transaction = await queryRunner.manager
```

Comment:
Why transaction, you don't write anything?

Reply:
User transaction...

Reply:
hidethepain...

Reply:
Gotta be honest I made a mistake few times before while reviewing my own stuff..